code/datums/components/crafting/guncrafting.dm

```diff
@@ -1,22 +1,165 @@
 //Gun crafting parts til they can be moved elsewhere
 
 // PARTS //
+/obj/item/weaponcrafting
+	name = "weapon part"
+	icon = 'icons/obj/improvised.dmi'
+	icon_state = "receiver"
 
 /obj/item/weaponcrafting/receiver
 	name = "modular receiver"
 	desc = "A prototype modular receiver and trigger assembly for a firearm."
-	icon = 'icons/obj/improvised.dmi'
-	icon_state = "receiver"
 
 /obj/item/weaponcrafting/stock
 	name = "rifle stock"
 	desc = "A classic rifle stock that doubles as a grip, roughly carved out of wood."
 	custom_materials = list(/datum/material/wood = MINERAL_MATERIAL_AMOUNT * 6)
-	icon = 'icons/obj/improvised.dmi'
 	icon_state = "riflestock"
 
-/obj/item/weaponcrafting/silkstring
+// ATTACHMENTS //
+
+/obj/item/weaponcrafting/attachment
+	//The following are buffs given to the weapon this is attached to
+	force = 0
+	sharpness = 0
+	bleed_force = 0
+	//if this changes the fire sound of the weapon, put it here
+	var/fire_sound = null
+	//If we can attach this to a bow
+	var/bow_suitable = TRUE
+	//If we can attach this to a standard gun, currently a useless var but will be important later
+	var/gun_suitable = TRUE
+	var/spread = 0
+	var/damage_multiplier = 0
+	var/speed_multiplier = 0
+	var/added_description = "<span class='info'>The item attached to this does absolutely nothing.</span>"
+
+// PRIMARY //
+// Primary attachments are meant to affect the mechanical aspect of a weapon, for now, since im working on bows, im using this to separate bow string from other attachments
+// knowing this can be usefull for the future when its expanded into other weapons
+
+/obj/item/weaponcrafting/attachment/primary/cablestring
+	name = "cable coil string"
+	desc = "A cable coil string to be used in weapon crafting."
+	icon_state = "cablestring"
+	damage_multiplier = 0.5
+	speed_multiplier = 2
+	gun_suitable = FALSE
+	//Half damage and Half speed
+	added_description = "<span class='info'>The bowstring is improvised out of cable. It looks rather weak.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/bamboostring
+	name = "bamboo fiber string"
+	desc = "A string made out of bamboo fiber to be used in weapon crafting."
+	icon_state = "bamboostring"
+	damage_multiplier = 0.8
+	speed_multiplier = 0.8
```

Comment:
```suggestion
	damage_multiplier = 0.9
	speed_multiplier = 0.8
```

Easily obtainable, but a good material to use. Good fast arrows.

Reply:
Done, buffed.